### PR TITLE
Memleak fixes

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes memory leaks found
* Fixes some API misuse

### CODE REVIEW

File: channels/rdpdr/client/rdpdr_main.c [hunk 1]
```diff
@@ -495,14 +500,22 @@ static UINT handle_hotplug(rdpdrPlugin* rdpdr)
...
+	for (i = 0; i < size; i++)
+		free (dev_array[size].path);
+
 	return rdpdr_send_device_list_announce_request(rdpdr, TRUE);
+
+cleanup:
+	for (i = 0; i < size; i++)
+		free (dev_array[size].path);
+
+	return error;
```
Comment (MartinHaimberger) [inline channels/rdpdr/client/rdpdr_main.c]: would nt it be better to remove the duplicated code above and use something like:

return error ? error : rdpdr_send_device_list_announce_request(rdpdr, TRUE);

File: channels/rdpdr/client/rdpdr_main.c [hunk 2]
```diff
@@ -468,17 +468,21 @@ static UINT handle_hotplug(rdpdrPlugin* rdpdr)
...
 
 			drive->Type = RDPDR_DTYP_FILESYSTEM;
 
-			drive->Path = _strdup(dev_array[i].path);
+			drive->Path = dev_array[i].path;
+			dev_array[i].path = NULL;
+
 			if (!drive->Path)
 			{
 				WLog_ERR(TAG, "_strdup failed!");
```
Comment (MartinHaimberger) [inline channels/rdpdr/client/rdpdr_main.c]: wrong comment because of removed _strdup

Comment (akallabeth) [inline channels/rdpdr/client/rdpdr_main.c]: @MartinHaimberger No, the ```_strdup``` just duplicated a string cleaned up later on. Setting the source to NULL prevents that and so no duplication is necessary.

**Comment by bmiklautz** [comment]
+1

**Comment by hardening** [comment]
+1

**Pull request merged**